### ISSUE TITLE
L019 adds a comma between the new qualified alias and the unqualified reference after L028

### ISSUE THREAD

**Issue by kevinsanz93** [sqlfluff/sqlfluff#3833]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Currently this block:
```
with a as (SELECT 'test' as key,'value' AS value)

select
key,
value

from a
where value is not null


```
will return 

```
with a as (select
    'test' as key
, 'value' as value)

select
    a.key
    a., value

from a
where a.value is not null

```

### Expected Behaviour

I would expect the return to be 
```
with a as (select
    'test' as key
, 'value' as value)

select
    a.key
    ,  a.value

from a
where a.value is not null

```

### Observed Behaviour

```
with a as (select
    'test' as key
, 'value' as value)

select
    a.key
    a., value

from a
where a.value is not null

```

### How to reproduce

Create a sql file with these contents: 
```
with a as (SELECT 'test' as key,'value' AS value)

select
key,
value

from a
where value is not null
```

then run sqlfluff fix on it given the included sqlFluff config.

Below are the rule violations from the lint logs

```
L:   1 | P:  12 | L010 | Keywords must be lower case.
L:   1 | P:  12 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
L:   1 | P:  29 | L029 | Keywords should not be used as identifiers.
L:   1 | P:  33 | L008 | Commas should be followed by a single whitespace unless
                       | followed by a comment.
L:   1 | P:  41 | L010 | Keywords must be lower case.
L:   1 | P:  44 | L029 | Keywords should not be used as identifiers.
L:   4 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 03]
L:   4 | P:   1 | L028 | Unqualified reference 'key' found in single table
                       | select.
L:   4 | P:   4 | L019 | Found trailing comma. Expected only leading.
L:   5 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 03]
L:   5 | P:   1 | L028 | Unqualified reference 'value' found in single table
                       | select.
L:   8 | P:   7 | L028 | Unqualified reference 'value' found in single table
                       | select.
L:   9 | P:   1 | L009 | Files must end with a single trailing newline.
```

### Dialect

snowflake

### Version

sqlfluff, version 1.3.0
```

### Configuration

[sqlfluff]
# verbose is an integer (0-2) indicating the level of log output
verbose = 0
# Turn off color formatting of output
nocolor = False
# Supported dialects https://docs.sqlfluff.com/en/stable/dialects.html
# Or run 'sqlfluff dialects'
dialect = snowflake
# One of [dbt|raw|jinja|python|placeholder]
templater = jinja
# Comma separated list of rules to check, default to all
rules = all
# Comma separated list of rules to exclude, or None
exclude_rules = L031
# The depth to recursively parse to (0 for unlimited)
recurse = 0
# Below controls SQLFluff output, see max_line_length for SQL output
output_line_length = 80
# Number of passes to run before admitting defeat
runaway_limit = 10
# Ignore errors by category (one or more of the following, separated by commas: lexing,linting,parsing,templating)
ignore = None
# Ignore linting errors found within sections of code coming directly from
# templated code (e.g. from within Jinja curly braces. Note that it does not
# ignore errors from literal code found within template loops.
ignore_templated_areas = True
# can either be autodetect or a valid encoding e.g. utf-8, utf-8-sig
encoding = autodetect
# Ignore inline overrides (e.g. to test if still required)
disable_noqa = False
# Comma separated list of file extensions to lint
# NB: This config will only apply in the root folder
sql_file_exts = .sql,.sql.j2,.dml,.ddl
# Allow fix to run on files, even if they contain parsing errors
# Note altering this is NOT RECOMMENDED as can corrupt SQL
fix_even_unparsable = False
# Very large files can make the parser effectively hang.
# This limit skips files over a certain character length
# and warns the user what has happened.
# Set this to 0 to disable.
large_file_skip_char_limit = 20000
# CPU processes to use while linting.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specifed_number.
# e.g. -1 means use all processors but one. 0  means all cpus.
processes = 1

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/indentation.html
indented_joins = False
indented_ctes = False
indented_using_on = True
indented_on_contents = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

# Some rules can be configured directly from the config common to other rules
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 80
indent_unit = space
comma_style = leading
single_table_references = qualified
unquoted_identifiers_policy = all

# Some rules have their own specific config
[sqlfluff:rules:L003]
hanging_indents = False

[sqlfluff:rules:L007]
operator_new_lines = after

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L011]
# Aliasing preference for tables
aliasing = explicit

[sqlfluff:rules:L012]
# Aliasing preference for columns
aliasing = explicit

[sqlfluff:rules:L013]
#Column expression without alias. Use explicit AS clause
allow_scalar = False

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L015]
#DISTINCT used with parentheses. NOTE: No Config Argument was documented

[sqlfluff:rules:L016]
# Line length
ignore_comment_lines = False
ignore_comment_clauses = False

[sqlfluff:rules:L017]
#Function name not immediately followed by parenthesis.

[sqlfluff:rules:L018]
#WITH clause closing bracket should be on a new line.

[sqlfluff:rules:L027]
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L026]
# References must be in FROM clause
# Disabled for some dialects (e.g. bigquery)
force_enable = False

[sqlfluff:rules:L028]
# References must be consistently used
# Disabled for some dialects (e.g. bigquery)
force_enable = False
single_table_references = qualified

[sqlfluff:rules:L029]
# Keywords should not be used as identifiers.
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L030]
# Function names
extended_capitalisation_policy = consistent
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L031]
# Avoid table aliases in from clauses and join conditions.
# Disabled for some dialects (e.g. bigquery)
force_enable = False

[sqlfluff:rules:L036]
wildcard_policy = single

[sqlfluff:rules:L038]
# Trailing commas
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]
# Null & Boolean Literals
capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses
forbid_subquery_in = both

[sqlfluff:rules:L047]
# Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

[sqlfluff:rules:L051]
# Fully qualify JOIN clause
fully_qualify_join_types = inner

[sqlfluff:rules:L052]
# Semi-colon formatting approach
multiline_newline = False
require_final_semicolon = False

[sqlfluff:rules:L054]
# GROUP BY/ORDER BY column references
group_by_and_order_by_style = consistent

[sqlfluff:rules:L057]
# Special characters in identifiers
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = False
additional_allowed_characters = None
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L059]
# Policy on quoted and unquoted identifiers
prefer_quoted_identifiers = False
ignore_words = None
ignore_words_regex = None
force_enable = False

[sqlfluff:rules:L062]
# Comma separated list of blocked words that should not be used
blocked_words = None
blocked_regex = None

[sqlfluff:rules:L063]
# Data Types
extended_capitalisation_policy = consistent
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L064]
# Consistent usage of preferred quotes for quoted literals
preferred_quoted_literal_style = consistent
# Disabled for dialects that do not support single and double quotes for quoted literals (e.g. Postgres)
force_enable = False

[sqlfluff:rules:L066]
min_alias_length = None
max_alias_length = None

```
### Are you willing to work on and submit a PR to address the issue?

- [] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Thanks for submitting this, yes I think this is a bug. I'm going to do some debugging to see what we can do with this. I can't think of a good reason about why this should be happening. Both rules are applying edits to `value`, but one should be creating something before, and the other should be creating something after.

**Comment by alanmcruickshank** [comment]
Confirmed on my side, here's a minimal way to recreate it:

# Command

```bash
sqlfluff fix test2.sql -d snowflake -x f -r L019,L028
```

# Config:
```ini
[sqlfluff:rules]
comma_style = leading
single_table_references = qualified

[sqlfluff:rules:L028]
force_enable = False
```

# Input
```sql
select
key,
value
from a
```

# Output
```sql
select
a.key
a., value
from a
```

# Diagnosis

This is an integration between L019 and L028. Both use a `create_before` fix, and unfortunately the fix to create the comma is being applied before we apply the fix to create the qualified table reference.

Looking at the implementation of both rules, I think this prompts a rethink of `L019` rather than a rethink of `L028`. The former currently crawls all `raw` segments and then uses a slightly convoluted mechanism to detect (and fix) leading/trailing commas appropriately.

I'm going to look at both `L019` and `L007` as part of #3673 fairly soon and I hope that the approach I'm going to take will resolve this issue too. 👍

**Comment by alanmcruickshank** [comment]
This is solved as part of #3847. It's included as part of the test suite.
